lib/printer.ts

```diff
@@ -308,6 +308,20 @@ function genericPrintNoParens(path: any, options: any, print: any) {
       return concat(parts);
     }
 
+    case "ChainExpression":
+      return concat([path.call(print, "expression"), ";"]);
```

Comment:
I believe this will insert a semicolon at the end of a genetically printed chain expression, which is wrong. Perhaps just use the value of `path.call(print, "expression")`? You could also add a test where adding a semicolon would fail, such as here: `a(b?.c)`.

Reply:
I made that change, thanks for the review! I tried to make a test, but I couldn't quite figure out how to get one working. Here's what I have, where am I going wrong?

```
  it("does not print semicolons for ChainExpression", function () {
    const node = b.chainExpression(
      b.expression("a(b?.c)"),
    );

    assert.strictEqual(recast.print(node).code, "a(b?.c)");
  });
```